File: TransCoder_tokenized_test_set_functions/java/SHORTEST_POSSIBLE_COMBINATION_TWO_STRINGS.java
static void printSuperSeq ( String a, String b ) {
  int m = a . length ( ), n = b . length ( );
  int [ ] [ ] dp = new int [ m + 1 ] [ n + 1 ];
  for ( int i = 0;
  i <= m;
  i ++ ) {
    for ( int j = 0;
    j <= n;
    j ++ ) {
      if ( i == 0 ) dp [ i ] [ j ] = j;
      else if ( j == 0 ) dp [ i ] [ j ] = i;
      else if ( a . charAt ( i - 1 ) == b . charAt ( j - 1 ) ) dp [ i ] [ j ] = 1 + dp [ i - 1 ] [ j - 1 ];
      else dp [ i ] [ j ] = 1 + Math . min ( dp [ i - 1 ] [ j ], dp [ i ] [ j - 1 ] );
    }
  }
  String res = "";
  int i = m, j = n;
  while ( i > 0 && j > 0 ) {
    if ( a . charAt ( i - 1 ) == b . charAt ( j - 1 ) ) {
      res = a . charAt ( i - 1 ) + res;
      i --;
      j --;
    }
    else if ( dp [ i - 1 ] [ j ] < dp [ i ] [ j - 1 ] ) {
      res = a . charAt ( i - 1 ) + res;
      i --;
    }
    else {
      res = b . charAt ( j - 1 ) + res;
      j --;
    }
  }
  while ( i > 0 ) {
    res = a . charAt ( i - 1 ) + res;
    i --;
  }
  while ( j > 0 ) {
    res = b . charAt ( j - 1 ) + res;
    j --;
  }
  System . out . println ( res );
}

File: TransCoder_tokenized_test_set_functions/python/SHORTEST_POSSIBLE_COMBINATION_TWO_STRINGS.py
def printSuperSeq ( a, b ) :
    m = len ( a )
    n = len ( b )
    dp = [ [ 0 ] * ( n + 1 ) for i in range ( m + 1 ) ]
    for i in range ( 0, m + 1 ) :
        for j in range ( 0, n + 1 ) :
            if not i :
                dp [ i ] [ j ] = j;
                
            elif not j :
                dp [ i ] [ j ] = i;
                
            elif ( a [ i - 1 ] == b [ j - 1 ] ) :
                dp [ i ] [ j ] = 1 + dp [ i - 1 ] [ j - 1 ];
                
            else :
                dp [ i ] [ j ] = 1 + min ( dp [ i - 1 ] [ j ], dp [ i ] [ j - 1 ] );
                
    index = dp [ m ] [ n ];
    
    res = [ "" ] * ( index )
    i = m
    j = n;
    
    while ( i > 0 and j > 0 ) :
        if ( a [ i - 1 ] == b [ j - 1 ] ) :
            res [ index - 1 ] = a [ i - 1 ];
            
            i -= 1
            j -= 1
            index -= 1
        elif ( dp [ i - 1 ] [ j ] < dp [ i ] [ j - 1 ] ) :
            res [ index - 1 ] = a [ i - 1 ]
            i -= 1
            index -= 1
        else :
            res [ index - 1 ] = b [ j - 1 ]
            j -= 1
            index -= 1
    while ( i > 0 ) :
        res [ index - 1 ] = a [ i - 1 ]
        i -= 1
        index -= 1
    while ( j > 0 ) :
        res [ index - 1 ] = b [ j - 1 ]
        j -= 1
        index -= 1
    print ( "" . join ( res ) )
